pkg/reconciler/v1alpha1/taskrun/resources/entrypoint.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// MountName is the name of the pvc being mounted (which
+	// will contain the entrypoint binary and eventually the logs)
+	MountName = "tools"
+
+	mountPoint                 = "/tools"
+	entrypointBin              = mountPoint + "/entrypoint"
+	entrypointJSONConfigEnvVar = "ENTRYPOINT_OPTIONS"
+)
+
+var toolsMount = corev1.VolumeMount{
+	Name:      MountName,
+	MountPath: mountPoint,
+}
+
+// GetCopyStep will return a Build Step (Container) that will
+// copy the entrypoint binary from the entrypoint image into the
+// volume mounted at mountPoint, so that it can be mounted by
+// subsequent steps and used to capture logs.
+func GetCopyStep() corev1.Container {
+	return corev1.Container{
+		Name:         "place-tools",
+		Image:        "gcr.io/k8s-prow/entrypoint",
+		Command:      []string{"/bin/cp"},
+		Args:         []string{"/entrypoint", entrypointBin},
+		VolumeMounts: []corev1.VolumeMount{toolsMount},
+	}
+}
+
+type entrypointArgs struct {
+	Args       []string `json:"args"`
+	ProcessLog string   `json:"process_log"`
+	MarkerFile string   `json:"marker_file"`
+}
+
+func getEnvVar(cmd, args []string) (string, error) {
+	entrypointArgs := entrypointArgs{
+		Args:       append(cmd, args...),
+		ProcessLog: "/tools/process-log.txt",
+		MarkerFile: "/tools/marker-file.txt",
+	}
+	j, err := json.Marshal(entrypointArgs)
+	if err != nil {
+		return "", fmt.Errorf("couldn't marshal arguments %q for entrypoint env var: %s", entrypointArgs, err)
+	}
+	return string(j), nil
+}
+
+// TODO: add more test cases after all, e.g. with existing env
+// var and volume mounts
```

Comment:
you can remove this comment for now i think @aaron-prindle , we can cover this in the follow up task ill make

Reply:
ping you can remove the TODO